File: src/main/java/net/exotia/exotiaprofanity/configuration/Configuration.java
package net.exotia.exotiaprofanity.configuration;

import org.bukkit.plugin.Plugin;
import java.net.URI;
import java.net.URISyntaxException;

public class Configuration {
    private final String apiUrl;
    private final String delimiter;
    private final boolean autoReconnect;

    public Configuration(Plugin plugin) {
        this.apiUrl = plugin.getConfig().getString("apiUrl");
        this.delimiter = plugin.getConfig().getString("delimiter");
        this.autoReconnect = plugin.getConfig().getBoolean("autoReconnect");
    }
    public URI getApiUrl() {
        try {
            return new URI(this.apiUrl);
        } catch (URISyntaxException exception) {
            throw new RuntimeException(exception);
        }
    }
    public String getDelimiter() {
        return this.delimiter;
    }
    public boolean isAutoReconnect() {
        return this.autoReconnect;
    }
}

File: src/main/java/net/exotia/exotiaprofanitykotlin/configuration/Configuration.kt
package net.exotia.exotiaprofanitykotlin.configuration

import org.bukkit.plugin.Plugin

class Configuration(plugin: Plugin) {
  val apiUrl: String
  val delimiter: String
  val autoReconnect: Boolean

  init {
    this.apiUrl = plugin.config.getString("apiUrl") ?: ""
    this.delimiter = plugin.config.getString("delimiter") ?: ""
    this.autoReconnect = plugin.config.getBoolean("autoReconnect")
  }
}
